crates/ruff_linter/src/rules/pylint/rules/unspecified_encoding.rs

```diff
@@ -70,13 +76,32 @@ pub(crate) fn unspecified_encoding(checker: &mut Checker, call: &ast::ExprCall)
         return;
     };
 
-    checker.diagnostics.push(Diagnostic::new(
+    let mut diagnostic = Diagnostic::new(
         UnspecifiedEncoding {
             function_name,
             mode,
         },
         call.func.range(),
-    ));
+    );
+
+    if checker.settings.target_version >= PythonVersion::Py310 {
+        diagnostic.set_fix(Fix::unsafe_edit(Edit::insertion(
+            ", encoding=\"locale\"".to_string(),
```

Comment:
Yes it would. `open()` and similar calls all have a required first argument AFAIK but it's still a pretty dirty and unsafe fix. 

I'd be interested in adding that utility for sure! Will set this to draft and use the autofix as a test case for the utility if there is no urgency behind it.

Reply:
Awesome, thanks @qdegraaf!